### PR TITLE
[IE VPU][DTS] shrink mask support for StridedSlice

### PR DESCRIPTION
**Pull request by nikita-kud**
#-42017

PR is based on https://github.com/openvinotoolkit/openvino/pull/3321

CI:
- [x] 2157 (except known issue #-45339)

### CODE REVIEW

**Comment by AndrewBakalinIntel** [comment]
Could you please add a few test cases to `dsr_strided_slice.cpp` to make sure the inference is correct?

File: inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_strided_slice.cpp [hunk 1]
```diff
@@ -132,8 +133,6 @@ void dynamicToStaticShapeStridedSlice(std::shared_ptr<ngraph::Node> target) {
     const auto all_zero = [](const std::vector<int64_t> & v) {return std::all_of(v.cbegin(), v.cend(), [](const int64_t & i){return i == 0;});};
     VPU_THROW_UNLESS(all_zero(stridedSlice->get_new_axis_mask()),
             "dynamicToStaticShapeStridedSlice transformation is not applicable for {}, new_axis_mask expected to be zeros", target);
-    VPU_THROW_UNLESS(all_zero(stridedSlice->get_shrink_axis_mask()),
-                "dynamicToStaticShapeStridedSlice transformation is not applicable for {}, shrink_axis_mask expected to be zeros", target);
     VPU_THROW_UNLESS(all_zero(stridedSlice->get_ellipsis_mask()),
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_strided_slice.cpp:136]: AFAIK, new_axis_mask and ellipsis_mask are also supported by Myriad plugin. 
Will it be expensive to support them in DTS as well?

Comment (Maxim-Doronin) [reply inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_strided_slice.cpp:136]: Discussed offline. Resolve

**Comment by nikita-kud** [comment]
@gladilov-gleb @AndrewBakalinIntel could you take a look?

**Review by AndrewBakalinIntel — APPROVED** [review]

**Review by Maxim-Doronin — APPROVED** [review]

**Pull request merged**